.circleci/config.yml

```diff
@@ -183,7 +183,31 @@ commands:
                 fi
               fi
             fi
-          name: "Halt job if code in '<< parameters.dir_names >>' directories are not changed"
+          name: "On PR branch, halt job if code in '<< parameters.dir_names >>' dir are not changed in last commit"
+
+  halt-test-if-e2e:
+    description: "Skip non-e2e test if e2e test files are the only changed files in last commit on PR and master branch"
```

Comment:
Why do the other tests care about e2e? Seems like a better separation of concerns would be "API unit tests should only run when API code changes", "UI unit tests should only run when UI code changes"